metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -249,6 +249,11 @@ def __init__(self, ctx: PipelineContext, config: BigQueryV2Config):
             config, self.report, self.profiling_state_handler
         )
 
+        # Global store of table identifiers for lineage filtering
+        self.table_refs: Set[str] = set()
+        # Maps (dataset_name, view_name) -> view_definition, for view lineage
+        self.view_definitions: Dict[Tuple[str, str], str] = {}
```

Comment:
I store as little information as possible to allow for us to do lineage after all projects' schemas have been processed. Turns out, all that lineage generation was using were each view's dataset_name, view_name, and view_definition.

Reply:
please see my comment above. I think we can store even less but that would need a refractor. I am open to handling this in follow up PR.